### PR TITLE
Upgrade smoke tested AGP versions

### PR DESCRIPTION
**Pull request by eskatos**
from 4.1.0 to 4.1.1
from 4.2.0-alpha16 to 4.2.0-beta03
add 7.0.0-alpha04

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test RFN

**Comment by eskatos** [comment]
@bot-gradle test RFN

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidPluginsSmokeTest.groovy [hunk 1]
```diff
@@ -135,7 +138,7 @@ class AndroidPluginsSmokeTest extends AbstractSmokeTest {
 
         where:
         [agpVersion, ide] << [
-            TestedVersions.androidGradle.toList(),
+            TestedVersions.androidGradle.toList().takeRight(2).take(1),
```
Comment (bamboo) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidPluginsSmokeTest.groovy:141]: ❌ I don't get this, could you maybe extract an explaining method of the intention here?

**Review by bamboo — CHANGES_REQUESTED** [review]

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidPluginsSmokeTest.groovy (hunk 1 above)
Comment (eskatos) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidPluginsSmokeTest.groovy]: Good catch. It's just a remnant of local testing. Removed

**Comment by eskatos** [comment]
@bot-gradle test RFN

**Review by bamboo — APPROVED** [review]
LGTM!

**Comment by wolfs** [comment]
@eskatos I merged the PR since I need the newer version to test AGP lint, which should now work with latest Gradle and AGP 4.2.0-beta03 again.

I wonder where we currently run the santa tracker AGP 7.0 tests. Do you have any insights into that?

**Comment by eskatos** [comment]
@wolfs `AbstractAndroidSantaTrackerSmokeTest` descendants should run it

But I only found runs from some intermediate state of this PR's branch
https://ge.gradle.org/scans/tests?search.relativeStartTime=P7D&search.rootProjectNames=gradle&search.timeZoneId=Europe/Paris&tests.container=org.gradle.smoketests.AndroidSantaTrackerKotlinCachingSmokeTest&tests.sortField=FAILED&tests.test=can%20cache%20Santa%20Tracker%20Kotlin%20Android%20application%20(agp%3D7.0.0-alpha04)&tests.unstableOnly=true

I assumed this job would run them as it uses Java 15 and AGP 7 requires Java >= 11
https://builds.gradle.org/buildConfiguration/Gradle_Check_SmokeTestsJava15?branch=master&mode=builds

But it seems they don't
https://ge.gradle.org/s/vid53pzeh4x5u/tests?search=Santa

Ah, that's because of
https://github.com/gradle/gradle/blob/b8a37f126a1f2f0f7507400f1e2c928c01ddbea1/subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidSantaTrackerKotlinCachingSmokeTest.groovy#L32-L33

I'll open another PR fixing that now that there's another condition to skip the test depending on the combination of Java and AGP versions.

**Comment by eskatos** [comment]
Here it is https://github.com/gradle/gradle/pull/15771

**Pull request merged**